syntax = "proto3";

package v1;

import "google/protobuf/timestamp.proto";

service CVProcessorService {
    rpc ProcessBatchPDF (BatchPDFProcessRequest) returns (BatchPDFProcessResponse);
    rpc FetchSummarizedPdfHistory(FetchSummarizedPdfHistoryRequest) returns (BatchPDFProcessResponse);
}

message BatchPDFProcessRequest {
    string bucket_name = 1;
    string batch_id = 2;
    string user_id = 3;
    repeated PDFFileInfo pdf_files = 4;
}

message PDFFileInfo {
    string file_name = 1;
    int64 size = 2;
    google.protobuf.Timestamp uploaded_at = 3;
}

message BatchPDFProcessResponse {
    string batch_id = 1;
    int32 total_files = 2;
    repeated PredictionResult predictions = 3;
}

message PredictionResult {
    string file_name = 1;
    CVPrediction prediction = 2;
}

message CVPrediction {
    repeated string name = 1;
    repeated string college_name = 2;
    repeated string degree = 3;
    repeated string graduation_year = 4;
    repeated string years_of_experience = 5;
    repeated string companies_worked_at = 6;
    repeated string designation = 7;
    repeated string skills = 8;
    repeated string location = 9;
    repeated string email_address = 10;
}

message FetchSummarizedPdfHistoryRequest {
    string user_id = 1;
}